sound/soc/sof/imx/imx8.c

```diff
@@ -368,10 +369,13 @@ static int imx8_get_bar_index(struct snd_sof_dev *sdev, u32 type)
 }
 
 static void imx8_ipc_msg_data(struct snd_sof_dev *sdev,
-			      struct snd_pcm_substream *substream,
+			      struct snd_sof_pcm_stream *sps,
 			      void *p, size_t sz)
 {
-	sof_mailbox_read(sdev, sdev->dsp_box.offset, p, sz);
+	if (!sps || !sdev->stream_box.size)
```

Comment:
same here, sps is not used.

Reply:
@iuliana-prodan this is because of the semantics `snd_sof_ipc_msg_data` has assigned for second parameter.

```void snd_sof_ipc_msg_data(struct snd_sof_dev *sdev, struct snd_sof_pcm_stream *sps, void *p, size_t sz)
```

when, sps is NULL it will return data from stream_box while when sps is not NULL is returned from dsp_box. 



```snd_sof_ipc_msg_data(sdev, NULL, &hdr, sizeof(hdr));
      -> imx8_ipc_msg_data(sdev, NULL, &hdr, sizeof(hdr));
``